### PR TITLE
Add Download Cypress Tests

### PR DESCRIPTION
**Pull request by davidcui1225**
*Issue #, if available:*
N/A
*Description of changes:*
* Added cypress tests for download actions (landing page, and PDF/PNG on context menu)
* Renamed `cypress/` to `.cypress/` to prevent Kibana restart on every change
* Added HTML ids to Reporting loading modal
* Updated snapshots 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: kibana-reports/.cypress/fixtures/example.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "Using fixtures to represent data",
+  "email": "hello@cypress.io",
+  "body": "Fixtures are a great way to mock data for responses to routes"
+}
```
Comment (zhongnansu) [inline kibana-reports/.cypress/fixtures/example.json]: where is this used?

File: kibana-reports/.cypress/integration/04-download.spec.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    // click Reporting in-context menu
+    cy.get('#downloadReport > span:nth-child(1) > span:nth-child(1)').click({ force: true });
+
+    // download PDF 
+    cy.get('#generatePDF > span:nth-child(1) > span:nth-child(2)').click({ force: true });
+
+    cy.get('#reportGenerationProgressModal');
+  });
+
+  it('Download png from in-context menu', () => {
```
Comment (zhongnansu) [inline kibana-reports/.cypress/integration/04-download.spec.ts:52]: Missing download csv?

**Review by zhongnansu — COMMENTED** [review]
leave few comments

File: kibana-reports/.cypress/fixtures/example.json (hunk 1 above)
Comment (davidcui1225) [reply kibana-reports/.cypress/fixtures/example.json]: Not used, this was added by accident when adding the `.cypress` directory, will remove

**Review by zhongnansu — APPROVED** [review]

**Review by joshuali925 — APPROVED** [review]
does downloading this way trigger the "select download directory" browser dialog?

**Comment by davidcui1225** [comment]
> does downloading this way trigger the "select download directory" browser dialog?

No, haven't seen that triggered.

**Pull request merged**